tests/test_streaming.py

```diff
@@ -246,3 +254,68 @@ def test_not_fast_enough_quad(
     else:
         # should be cancelled mid-streaming
         assert results is None
+
+
+@tractor_test
+async def test_respawn_consumer_task(
+    arb_addr,
+    spawn_backend,
+    loglevel,
+):
+    """Verify that ``._portal.StreamReceiveChannel.shield_channel()``
+    sucessfully protects the underlying IPC channel from being closed
+    when cancelling and respawning a consumer task.
+
+    This also serves to verify that all values from the stream can be
+    received despite the respawns.
+
+    """
+    stream = None
+
+    async with tractor.open_nursery() as n:
+
+        stream = await(await n.run_in_actor(
```

Comment:
Btw, what do y'all think about a `ActorNursery.stream_from_actor()` sugar for this pattern?

Reply:
In theory you can actually just call `Portal.result()` before returning from `.run_in_actor()` but it means each call is then blocking and you'd have to spawn multiple (one-off) actors using tasks. Not sure if this is a problem and maybe will get hashed out with the discussion in #66.